projects/plugins/jetpack/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -982,8 +982,12 @@ public function update_data( $request ) {
 
 				case 'jetpack_subscriptions_reply_to':
 					// If option value was the same, consider it done.
-					$sub_value = in_array( $value, array( 'no-reply', 'author' ), true ) ? $value : 'no-reply';
-					$updated   = (string) get_option( $option ) !== (string) $sub_value ? update_option( $option, $sub_value ) : true;
+					require_once JETPACK__PLUGIN_DIR . 'modules/subscriptions/class-settings.php';
+					$sub_value = Automattic\Jetpack\Modules\Subscriptions\Settings::is_valid_reply_to( $value )
+						? $value
+						: Automattic\Jetpack\Modules\Subscriptions\Settings::get_default_reply_to();
+
+						$updated = (string) get_option( $option ) !== (string) $sub_value ? update_option( $option, $sub_value ) : true;
```

Comment:
I think core does this optimization for you for you already!

Reply:
It does but we need to know about this value early so that we can say correctly if the value was updated.